Dockerfile

```diff
@@ -1,56 +1,53 @@
 # syntax=docker/dockerfile:1.4
 
 ####################################################################################################
-## Prepare Workspace
+## Build Packages
 
-FROM node:18-alpine AS workspace
-WORKDIR /workspace
+FROM node:18-alpine AS builder
+WORKDIR /directus
+
+ENV NODE_OPTIONS=--max-old-space-size=8192
 
-COPY pnpm-lock.yaml .
 RUN corepack enable && corepack prepare pnpm@7.30.0 --activate
+
+COPY pnpm-lock.yaml .
 RUN pnpm fetch
 COPY . .
 RUN pnpm install --recursive --offline --frozen-lockfile
 
-####################################################################################################
-## Create Deployment
-
-FROM workspace AS pruned
-WORKDIR /workspace
-ENV NODE_OPTIONS=--max-old-space-size=8192
-
-RUN pnpm --recursive --workspace-concurrency=1 run build
-RUN pnpm --filter directus deploy --prod pruned
-RUN cd pruned \
+RUN : \
+	&& pnpm --recursive --workspace-concurrency=1 run build \
+	&& pnpm --filter directus deploy --prod dist \
+	&& cd dist \
 	&& pnpm pack \
 	&& tar -zxvf *.tgz package/package.json \
 	&& mv package/package.json package.json \
 	&& rm -r *.tgz package \
-	&& mkdir database extensions uploads
+	&& mkdir -p data/database data/extensions data/uploads \
+	;
 
 ####################################################################################################
 ## Create Production Image
 
-FROM node:18-alpine
+FROM node:18-alpine AS runtime
 
 WORKDIR /directus
 
 EXPOSE 8055
 
-ENV DB_CLIENT="sqlite3"
-ENV DB_FILENAME="/directus/database/database.sqlite"
-ENV EXTENSIONS_PATH="/directus/extensions"
-ENV STORAGE_LOCAL_ROOT="/directus/uploads"
-ENV NODE_ENV="production"
-ENV NPM_CONFIG_UPDATE_NOTIFIER="false"
-
-COPY --from=pruned --chown=node:node /workspace/pruned .
+ENV \
+	DB_CLIENT="sqlite3" \
+	DB_FILENAME="/directus/data/database/database.sqlite" \
+	EXTENSIONS_PATH="/directus/data/extensions" \
+	STORAGE_LOCAL_ROOT="/directus/data/uploads" \
+	NODE_ENV="production" \
+	NPM_CONFIG_UPDATE_NOTIFIER="false"
 
-VOLUME /directus/database
-VOLUME /directus/extensions
-VOLUME /directus/uploads
```

Comment:
Adding `VOLUME` to the Dockerfile is OK, but adds to build size and complexity. Leaving the volume declaration to the runtime removes the creation of volume artifacts during build (which must *still* be linked at runtime anyway!).

If for some reason, we insist on keeping this here, this should be reduced to a single "data" volume or combined into a single layer.

Reply:
TIL!